topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/stream/SerDesConfig.java

```diff
@@ -0,0 +1,146 @@
+ * - Each input has a pair of key/value deserialisers
+ * - Each output has a pair of key/value serialisers
+ * In general, the keys are not used so the choice of serialiser is not important
+ */
+public final class SerDesConfig {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final String SERIALIZATION_FAILED_MESSAGE = "Failed to serialize ";
+    private static final String DESERIALIZATION_FAILED_MESSAGE = "Failed to deserialize ";
+
+    private SerDesConfig() {
+        // Static collection of objects, class should never be instantiated
+    }
+
+    /**
+     * Kafka key serialiser for upstream messages coming in as input
+     * Used by the Rest Controller to insert requests onto the topic
+     *
+     * @return an appropriate key serialiser for the topic's message content
+     */
+    public static Serializer<String> topicOffsetRequestKeySerializer() {
+        return new StringSerializer();
+    }
+
+    /**
+     * Kafka value serialiser for upstream messages coming in as input
+     * Used by the Rest Controller to insert requests onto the topic
+     *
+     * @return an appropriate value serialiser for the topic's message content (TopicOffsetRequest)
+     */
+    public static Serializer<TopicOffsetRequest> topicOffsetRequestValueSerializer() {
+        return (String ignored, TopicOffsetRequest topicOffsetRequest) -> {
+            try {
+                return MAPPER.writeValueAsBytes(topicOffsetRequest);
+            } catch (IOException e) {
+                throw new SerializationFailedException(SERIALIZATION_FAILED_MESSAGE + topicOffsetRequest.toString(), e);
+            }
+        };
+    }
+
+    /**
+     * Kafka key deserialiser for upstream messages coming in as input
+     *
+     * @return an appropriate key deserialiser for the topic's message content
+     */
+    public static Deserializer<String> topicOffsetRequestKeyDeserializer() {
+        return new StringDeserializer();
+    }
+
+    /**
+     * Kafka value deserialiser for upstream messages coming in as input
+     *
+     * @return an appropriate value deserialiser for the topic's message content (TopicOffsetRequest)
+     */
+    public static Deserializer<TopicOffsetRequest> topicOffsetRequestValueDeserializer() {
+        return (String ignored, byte[] topicOffsetRequest) -> {
+            try {
+                return MAPPER.readValue(topicOffsetRequest, TopicOffsetRequest.class);
+            } catch (IOException e) {
+                throw new SerializationFailedException(DESERIALIZATION_FAILED_MESSAGE + new String(topicOffsetRequest, Charset.defaultCharset()), e);
+            }
+        };
+    }
+
+    /**
+     * Kafka key serialiser for downstream messages going out as output
+     *
+     * @return an appropriate key serialiser for the topic's message content
+     */
+    public static Serializer<String> topicOffsetRequestOffsetKeySerializer() {
```

Comment:
Again, maybe use the name of the topic in the method
```suggestion
    public static Serializer<String> maskedResourceOffsetKeySerializer() {
```